core/src/com/projectkorra/projectkorra/object/HorizontalVelocityTracker.java

```diff
@@ -86,33 +92,36 @@ public void update() {
 
 		final Vector diff = this.thisVelocity.subtract(this.lastVelocity);
 
-		final List<Block> blocks = GeneralMethods.getBlocksAroundPoint(this.entity.getLocation(), 1.5);
+		if (this.thisVelocity.length() < this.lastVelocity.length()) {
+			if ((diff.getX() > 0 || diff.getX() < 0) || (diff.getZ() > 0 || diff.getZ() < 0)) {
```

Comment:
is there any reason this isn't just diff.getX() != 0, same for getZ. Also the if can be merged with the above length check

Reply:
Problems with floating point precision, I believe. The code never got past that check.

Reply:
how is diff.getX() > 0 || diff.getX() < 0 not the same as diff.getX() != 0, weirdddd, and you've tried that yourself?

Reply:
Yeah so `diff.getX()` and `diff.getZ()` don't return integers, they would return some real number like 0.0000001 or -0.0000001 (not those exact numbers, just a floating-point representation) and so `!= 0` is not going to pass because of floating-point approximation. And yes, I ran a bunch of debugs through that entire condition block. If I remember correctly, I changed `diff.getX() > 1 || diff.getX() < -1` (and same for `#getZ`) to this because I found that they actually return numbers `0 < x < R` or `-R < x < 0`; 1 was too high a bound and so those conditions never got passed.

I.e., `#getX`/`#getZ` would more often return 1/10^n * k, or a number looking like 0.000000..., and not 1.000000..., so I lowered the bound.

Reply:
Actually. I misread, this is correct. They are logically the same, yes, so I believe it could just simply be `diff.getX() != 0` and same for `diff.getZ()`.